components/zcash_protocol/src/value.rs

```diff
@@ -229,6 +230,24 @@ impl Mul<usize> for ZatBalance {
 #[derive(Clone, Copy, Debug, PartialEq, PartialOrd, Eq, Ord)]
 pub struct Zatoshis(u64);
 
+/// A struct that provides both the dividend and remainder of a division operation.
```

Comment:
We already have a dependency on num-integer which has https://docs.rs/num-integer/latest/num_integer/fn.div_rem.html

Reply:
We would have to implement all the `num_integer::Integer` trait methods for `Zatoshis` to take advantage of that; honestly I don't think that's worth it.

Reply:
Is it worth following the signature of `div_rem` (or `div_floor_mod`) anyway?

Reply:
A struct is better than a tuple.

Reply:
Also, dividing a `Zatoshis` value by another `Zatoshis` makes no sense.

Reply:
Well it does, but the quotient would be dimensionless rather than `Zatoshis`, so yes it's not what we want here.

Reply:
quotient! I had been trying to remember that word, that's what this struct should use, not dividend.